cmake/IgnConfigureBuild.cmake

```diff
@@ -255,7 +255,9 @@ macro(ign_configure_build)
       else()
 
         set(skip_msg "Skipping the component [${component}]")
-        if(${component}_MISSING_DEPS)
+        if(SKIP_${component})
+          ign_string_append(skip_msg "by user request")
```

Comment:
Yep, the weird space here is necessary because cmake message formatting rules [are wild](https://stackoverflow.com/questions/50447052/cmake-multiline-message-with-fatal-error/51035045#51035045). If we don't include this space, then the warning will come out looking very goofy because it will be viewed as an "unformatted" string, and then cmake will apply some weird markdown interpretation rules to it.

Reply:
Oh wait, I was wrong about this case. I added spaces to the front of too many strings while trying to fix the formatting weirdness. I'll fix this spacing.

Reply:
Okay sorry I confused myself yet again. Somehow my brain thought the suggestion was to remove a space rather than add a space.

So here's what's actually happening: `ign_string_append` will automatically put in a space when appending `"by user request"` to the end of whatever `skip_msg` happens to contain. That's why we don't need to start this string with a space. This is completely unrelated to the message formatting rules, so please disregard my earlier comment about that.

Reply:
Ahh I didn't realize this would be so involved, thanks for iterating!

Reply:
No worries, I confused myself twice in quick succession because I was looking at it too quickly, but there wasn't any trouble.